blog/a-git-repository-inside-an-arch-linux-package.md

```diff
@@ -0,0 +1,109 @@
+# A Git repository inside an Arch Linux package
+
+Create a package from a Git repository can be used for testing the latest source code from the development branch, and catch issues before made a new release using a Git tag. Also you can use another Version Control System (VCS) like SVN, but we're going to focus in Git today.
+
+Some packages in the official Arch Linux repository like `konsole`, has a VCS package version called `konsole-git` in the AUR. The suffix `-git`, `-svn`, `-hg`, etc. is a convention to let the package users know that this is a VCS package.
+
+## The providing and conflicts
+
+In our `PKGBUILD` we can define a `provides` array variable to let `pacman` know which additional or virtual packages our package provides. E.g. `konsole-git` provides `konsole`.
+
+```bash
+pkgname=konsole-git
+...
+provides=('konsole')
+```
+
+This doesn't mean a conflict unless we define the `conflicts` variable too.
+
+```bash
+conflicts=('konsole')
+```
+
+### The implicit conflict
+
+* `nodejs` implicitly provides `nodejs` as the `pkgname` itself.
+* `node-lts-erbium` provides `nodejs` and conflicts with `nodejs`.
+* `nodejs-lts-dubnium` provides `nodejs` and conflicts with `nodejs`, but does not need to explicitly conflict with `node-lts-erbium` since packages providing the same feature are implicitly in conflict.
+
+## The pkgver as a function
+
+The VCS packages always compile the latest version in the repository. This is achieved using `pkgver()` function as the following:
+
+```bash
+pkgver() {
+  cd ${pkgname}
+  printf "r%s.%s" "$(git rev-list --count HEAD)" "$(git rev-parse --short HEAD)"
+}
+```
+
+Like every bash function, the return value is what the function prints to the `stdout` (standard output).
+
+The `git rev-list --count HEAD` command prints the count of commits in the branch, and `git rev-parse --short HEAD` prints the SHA-1 of the latest commit in short format (7 chars). The `printf "r%s.%s"` just format both values to something like `r155.083ffa8`.
+
+```bash
+makepkg
+```
+
+The `makepkg` will run the `pkgver()` function and will update the `PKGBUILD` value with the latest commit version value:
+
+```bash
+pkgver=r155.083ffa8
+```
+
+You can find other ways to define a VCS package version in the [Arch Wiki](https://wiki.archlinux.org/index.php/VCS_package_guidelines#Git).
+
+## The package source
+
+The `source` variable in the `PKGBUILD` just need to indicate the VCS (`git`) to download the sources, and we can't validate the checksum for a repository, then it should be skipped.
```

Comment:
```suggestion
The `source` variable in the `PKGBUILD` just needs to indicate the VCS (`git`) to download the sources, and since we can't validate the checksum for a repository, it should be skipped.
```